src/System.Data.SqlClient/src/System/Data/SqlClient/SqlClientDiagnosticListenerExtensions.cs

```diff
@@ -205,5 +217,182 @@ public static void WriteConnectionCloseError(this DiagnosticListener @this, Guid
                     });
             }
         }
+
+        public static Guid WriteTransactionCommitBefore(this DiagnosticListener @this, IsolationLevel isolationLevel, SqlConnection connection, [CallerMemberName] string operation = "")
```

Comment:
@saurabh500 But than it won't follow the convention of the method above: WriteConnectionCloseBefore. 
Sure I can amend the commit and change to WriteBeforeTransactionCommit if you think it is better.